### ISSUE TITLE
How to improve exploration?

### ISSUE THREAD

**Issue by ahmad-hl** [godka/Pensieve-PPO#11]
What is the parameter to enable better exploration in PPO?
Is it [EPS  ](https://github.com/godka/Pensieve-PPO/blob/3eed104b2ba9ef528f1dea43be0e835cd21560d7/src/ppo2.py#L13)?
How to control exploration and exploitation in PPO?
Do you recommend a good approach for this purpose?

**Comment by godka** [comment]
Nop, it's not EPS. It's controlled by **self. _entropy_weight**.
We have leveraged three different methods to encourage exploration till now.

In the first version, we follow Hongzi's advice and use the fixed "entropy ladder".
https://github.com/godka/Pensieve-PPO/blob/24ca724d20f318ef49019a07aaf8404f117b3dcd/src/ppo2.py#L106

While considering that it requires a long time to converge, we propose AEWD (adaptive entropy weight decay) in the second version.

https://github.com/godka/Pensieve-PPO/blob/ad4d194867cf63b2c69cb748d81a7b52e6d03ec8/src/train.py#L137

The key idea is to drop down the entropy weight if the performance has no longer been improved on the validation set.
For more information please check one of our papers. [Zwei paper](https://ieeexplore.ieee.org/document/9371417)

The final version uses the SOTA named **auto entropy method**, which is proposed by MSRA([paper](https://arxiv.org/abs/2003.13590)): To ensure stable exploration, we dynamically adjust entropy weight to increase/decrease the entropy term if the entropy of our policy is smaller/larger than the target (0.1) in a recent period.

**Comment by ahmad-hl** [comment]
Amazing and very inspiring work
Thank you for the informative details and the contribution
